tests/Feature/api/v1/Room/FileTest.php

```diff
@@ -121,39 +147,59 @@ public function testViewFiles()
 
         // Testing guests with access code
         $this->withHeaders(['Access-Code' => $this->room->accessCode])->getJson(route('api.v1.rooms.files.get', ['room'=>$this->room]))
-            ->assertSuccessful();
+            ->assertSuccessful()
+            ->assertJsonCount(0, 'data.files');
+        $this->flushHeaders();
 
         // Testing users with access code
         $this->actingAs($this->user)->withHeaders(['Access-Code' => $this->room->accessCode])->getJson(route('api.v1.rooms.files.get', ['room'=>$this->room]))
-            ->assertSuccessful();
+            ->assertSuccessful()
+            ->assertJsonCount(0, 'data.files');
+        $this->flushHeaders();
 
         // Testing member
         $this->room->members()->attach($this->user, ['role'=>RoomUserRole::USER]);
         $this->actingAs($this->user)->getJson(route('api.v1.rooms.files.get', ['room'=>$this->room]))
-            ->assertSuccessful();
+            ->assertSuccessful()
+            ->assertJsonCount(0, 'data.files');
 
         // Testing moderator member
         $this->room->members()->sync([$this->user->id,['role'=>RoomUserRole::MODERATOR]]);
         $this->actingAs($this->user)->getJson(route('api.v1.rooms.files.get', ['room'=>$this->room]))
-            ->assertSuccessful();
+            ->assertSuccessful()
+            ->assertJsonCount(0, 'data.files');
+
+        // Testing co-owner member
+        $this->room->members()->sync([$this->user->id,['role'=>RoomUserRole::CO_OWNER]]);
+        $this->actingAs($this->user)->getJson(route('api.v1.rooms.files.get', ['room'=>$this->room]))
+            ->assertSuccessful()
+            ->assertJsonFragment(['filename'=>$this->file_valid->name]);
 
         // Testing owner
         $this->actingAs($this->room->owner)->getJson(route('api.v1.rooms.files.get', ['room'=>$this->room]))
             ->assertSuccessful()
             ->assertJsonFragment(['filename'=>$this->file_valid->name]);
 
+        // Remove membership roles and test with view all permission
+        $this->room->members()->sync([]);
```

Comment:
What about the rooms.manage permission. Add a test for it!

Reply:
As the rooms.viewAll permission is inherited by the rooms.manage permission, I don't see a reason to check it again. I did not add these redundant tests in a few places. To check that the permission inheritance is working, there is the 'testPermissionInheritances' in tests/Feature/api/v1/Room/RoomTest.php